### PR TITLE
Add KeyBy 

### PR DESCRIPTION
**Pull request by thetnaingtn**
Add support for lodash's keyBy utility

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -535,6 +536,28 @@ m2 := lo.MapValues[int, int64, string](m, func(x int64, _ int) string {
 // map[int]string{1: "1", 2: "2", 3: "3"}
 ```
 
+### KeyValue
```
Comment (samber) [inline README.md:539]: KeyValue -> KeyBy

File: README.md [hunk 2]
```diff
@@ -78,6 +78,7 @@ Supported helpers for maps:
 - FromEntries
 - Assign (merge of maps)
 - MapValues
+- KeyBy
```
Comment (samber) [inline README.md:81]: Could you move it into the "Slice" section, please?

Also from map.go to slice.go 🙏

File: map.go [hunk 3]
```diff
@@ -70,3 +70,14 @@ func MapValues[K comparable, V any, R any](in map[K]V, iteratee func(V, K) R) ma
 
 	return result
 }
+
+func KeyBy[T any, V comparable](entries []T, iteratee func(T) V) map[V]T {
```
Comment (samber) [inline map.go:74]: "V" to "K" please

It will be more understandable.

File: map_test.go [hunk 4]
```diff
@@ -6,6 +6,7 @@ import (
 	"testing"
 
 	"github.com/stretchr/testify/assert"
+	"golang.org/x/tools/go/expect"
```
Comment (samber) [inline map_test.go:9]: i'm pretty sure you can remove this line

Comment (samber) [inline map_test.go:9]: i'm pretty sure you can remove this line

**Review by samber — CHANGES_REQUESTED** [review]

**Comment by samber** [comment]
Oops, it is the same helper than lo.ToMap actually.

Can you rename ToMap to KeyBy please ?

**Comment by thetnaingtn** [comment]
> Oops, it is the same helper than lo.ToMap actually.
> 
> Can you rename ToMap to KeyBy please ?

Sure thing. Now, I removed duplicate KeyBy and renamed ToMap utility to KeyBy.  @samber

**Comment by samber** [comment]
Thanks for your help and sorry for the wrong name.

Let's merge ;)

**Pull request merged**